utils/defaulter.go

```diff
@@ -44,13 +45,20 @@ func getKongDefaulter(opts DefaulterOpts) (*Defaulter, error) {
 	if err := d.populateDefaultsFromInput(opts.KongDefaults); err != nil {
 		return nil, err
 	}
+
+	// TODO check if it's OK to do the validation only for the user input
+	err := d.validateKongDefaults()
```

Comment:
Since this is about checking whether the user is setting restricted fields defaults, what do you think of moving this call within the `populateDefaultsFromInput` method?

Reply:
+1, I refactored **validateKongDefaults** and moved the check in **populateDefaultsFromInput**